<launch>

        <!-- RLE Parameters -->
        <param name="/road_layout_estimation/resampling_interval"               value="4"/>
        <param name="/road_layout_estimation/particles_number"                  value="30"/>
        <param name="/road_layout_estimation/rle_frequency"                     value="0.5"/>     <!-- RLE loop frequency HZ -->

        <!--MAX VALUES HERE ARE 10 [10 is set in the cfg file] -->
        <param name="/road_layout_estimation/street_distribution_alpha"        value="1"/>     <!--these values should be greater than zero-->
        <param name="/road_layout_estimation/angle_distribution_alpha"         value="1"/>
        <param name="/road_layout_estimation/roadState_distribution_alpha"     value="1"/>
        <param name="/road_layout_estimation/roadLane_distribution_alpha "     value="1"/>

        <!-- PARAMETERS OSM DISTANCE COMPONENT [check max values in the .cfg file -->
        <param name="/road_layout_estimation/street_distribution_sigma"     value="1.5"/>
        <param name="/road_layout_estimation/angle_distribution_sigma "     value="0.15"/>

        <!--State6DOF ***ABSOLUTE***-->
        <param name="/road_layout_estimation/propagate_translational_absolute_vel_error_x"              value="0.18" /> <!-- value="0.01" />-->
        <param name="/road_layout_estimation/propagate_translational_absolute_vel_error_y"              value="0.01" /> <!-- value="0.01" />-->
        <param name="/road_layout_estimation/propagate_translational_absolute_vel_error_z"              value="0.0" /> <!-- value="0.01" />-->
        <param name="/road_layout_estimation/propagate_rotational_absolute_vel_error"                   value="0.03" /> <!-- value="0.05" />-->
                                                                                                                        <!--                -->
        <!--#State6DOF ***PERCENTAGE***-->                                                                              <!--                -->
        <param name="/road_layout_estimation/propagate_translational_percentage_vel_error_x"            value="0.1" /> <!-- value="0.01" />-->
        <param name="/road_layout_estimation/propagate_translational_percentage_vel_error_y"            value="0.01" /> <!-- value="0.01" />-->
        <param name="/road_layout_estimation/propagate_translational_percentage_vel_error_z"            value="0.0" /> <!-- value="0.01" />-->
        <param name="/road_layout_estimation/propagate_rotational_percentage_vel_error"                 value="0.05" /> <!-- value="0.05" />-->


        <!-- ARGUMENTS -->
            <!-- DEFAULT behavior, append bagfile:=kitti_XX to BAG in the command line-->
            <arg name="bagfile" default="2011_09_26_drive_0059_sync" />
            <arg name="camera" default="/kitti_stereo" />                                                               <!-- The namespace where images are published -->
            <arg name="publish_rate" default="0.1" />
            <arg name="rviz" default="$(find road_layout_estimation)/rviz/visual_odometry_kitti_viso.rviz" />
            <arg name="map" default="$(find ira_open_street_map)/maps/$(arg bagfile).osm" />

        <!--disabled since added inside the sgbm-building-detector node-->
        <!--<include file="$(find ira_open_street_map)/launch/$(arg bagfile).launch" />-->


        <!-- STATIC TRANFORMS -->
            <!-- Static transform fixing Viso moving frame axis -->
                <node pkg="tf" type="static_transform_publisher" name="visual_odometry_camera_frame_to_visual_odometry_car_frame"
                        args="0 0 0 0 -1.570796 1.570796 visual_odometry_camera_frame visual_odometry_car_frame 50" />
                <!-- Transform from local_map (osm package) to road_layout_estimation fixed frame-->
                <node pkg="tf" type="static_transform_publisher" name="odom_to_visual_odometry_odom_x_forward"
                        args="0 0 0 -1.570796 0 -1.570796 visual_odometry_odom_x_forward odom 50" />
                <!-- Transform from local_map (osm package) to road_layout_estimation fixed frame -->
                <node pkg="tf" type="static_transform_publisher" name="local_map_to_odom"
                        args="0 0 0 0 0 0 local_map visual_odometry_odom_x_forward 50" />
                <node pkg="tf" type="static_transform_publisher" name="dummy_transform"
                        args="0 0 0 0 0 0 base_link visual_odometry_camera_frame 50" />
			
        <!--disabled since added inside the sgbm-building-detector node-->
        <!-- OSM QUERY NODE
            <node
                    pkg="ira_open_street_map"
                    type="osm_query_node"
                    name="osm_query_node"
                    output="screen"
                    args="$(arg map)"
                    required="true"
            />
        -->


        <!-- to debug in QtCreator, run the road_layout_estimation node with
             /stereo_odometer/odometry 2011_10_03_drive_0042_sync as arguments and
             disable the following node from the launcher -->
        <!-- Launch ROAD LAYOUT ESTIMATION node-->
        <node
		pkg="road_layout_estimation" type="road_layout_estimation" name="road_layout_estimation"
                required="true"
		output="screen"
                args="/stereo_odometer/odometry $(arg bagfile)"
        />

        <!-- VISUALIZATION STUFF -->
            <!-- Launch RViz-->
            <node pkg="rviz" type="rviz" name="rviz" respawn="true" args="-d $(arg rviz)" output="screen">
            </node>

        <!-- Launch RQT reconfigure
        <node pkg="rqt_reconfigure" type="rqt_reconfigure" name="rqt_reconfigure" respawn="true"/> -->

        <!-- Launch ROAD LANE DETECTOR  ** IRALAB**
            <node
                    pkg="road_lane_detection" type="bev_lane_detection_node" name="bev_lane_detection_node"
                    required="true"
                    output="screen"
            />
         -->

        <!-- Launch ROAD LANE DETECTOR  ** ISIS LAB**
        <include file="$(find isis_line_detector)/launch/linedetector.launch" />-->





</launch>
